tests/test_fluid_simulation.py

```diff
@@ -113,3 +108,315 @@ def test_fluid_simulation_phases(self, fluent_simulation):
+        assert result.index.mesh_index.location == "cells"
+        assert result._core_object[0].location == post.locations.elemental
+
+    def test_results_fluent_averaging_from_elemental_faces(self, fluent_simulation):
+        print(fluent_simulation)
+        # ######## ElementalFaces Result #################
+        # Request on None
+        result = fluent_simulation.static_pressure()
+        assert result.index.mesh_index.location == "cells"
+        assert result._core_object[0].location == post.locations.elemental
+
+        # Request on nodes
+        result = fluent_simulation.static_pressure(location=post.locations.nodal)
+        assert result.index.mesh_index.location == post.locations.nodal
+        assert result._core_object[0].location == post.locations.nodal
+        result = fluent_simulation.static_pressure_on_nodes()
+        assert result.index.mesh_index.location == post.locations.nodal
+        assert result._core_object[0].location == post.locations.nodal
+
+        # Request on faces (requires filter-out of cell zones)
+        with pytest.raises(
+            ValueError,
+            match="Querying an ElementalAndFaces result on "
+            "faces currently requires the use of face zone ids",
+        ):
+            _ = fluent_simulation.static_pressure(location=post.locations.faces)
+        # print(result)
+        # assert result.index.mesh_index.location == post.locations.faces
+        # assert result._core_object[0].location == post.locations.faces
+        # result._fc[0].plot()
+        with pytest.raises(
+            ValueError,
+            match="Querying an ElementalAndFaces result on "
+            "faces currently requires the use of face zone ids",
+        ):
+            _ = fluent_simulation.static_pressure_on_faces()
+        # print(result)
+        # assert result.index.mesh_index.location == post.locations.faces
+        # # assert result._core_object[0].location == post.locations.faces
+        # result.plot()
+
+        # Request on cells (requires filter-out of face zones)
+        result = fluent_simulation.static_pressure(location=post.locations.elemental)
+        assert result.index.mesh_index.location == "cells"
+        assert result._core_object[0].location == post.locations.elemental
+        result = fluent_simulation.static_pressure_on_cells()
+        assert result.index.mesh_index.location == "cells"
+        assert result._core_object[0].location == post.locations.elemental
+
+    def test_results_cfx_cross_locations_on_nodes(self, cfx_simulation):
+        result = cfx_simulation.density_on_nodes(node_ids=cfx_simulation.mesh.node_ids)
+        assert result.index.mesh_index.location == post.locations.nodal
+        ref = """
+  results     RHO (kg*m^-3)           
+  set_ids                 1           
+    phase Water at 25 C (2) Copper (3)
+ node_ids                             
+        1        9.9700e+02           
```

Comment:
Maybe density is not the best result because this seems to be an incompressible simulation (density is constant), so we will not be able to catch bugs regarding wrong scoping transposition)

Reply:
ok, I'll change for something else